### PR TITLE
FTP-1383:Design-systems-components-buttons

### PR DESCRIPTION
**Pull request by tushar8184**
Screenshot 2022-10-03 at 12 23 21 PM

### CODE REVIEW

File: components/button/Button.test.tsx [hunk 1]
```diff
@@ -1,14 +1,13 @@
 import { render, screen } from '@testing-library/react';
```
Comment (skippednote) [inline components/button/Button.test.tsx:1]: Please revert this file

File: components/button/Button.test.tsx [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+    render(<button>Test data</button>);
     const button = screen.getByRole('button');
     expect(button).toBeInTheDocument();
   });
   it('renders the correct child data', () => {
-    render(<Button>Test data</Button>);
+    render(<button>Test data</button>);
     const button = screen.getByRole('button');
     expect(button).toHaveTextContent(/test data/i);
   });
```
Comment (skippednote) [inline components/button/Button.test.tsx:13]: Add more tests to check all variants and other props

File: components/button/Button.tsx [hunk 3]
```diff
@@ -1,7 +1,50 @@
 import React from 'react';
+import { ButtonType } from '../../@types/button.types';
 
-const Button: React.FC<{ children: React.ReactNode }> = ({ children }) => {
-  return <button>{children}</button>;
+const Button = ({
+  variant,
+  text = '',
```
Comment (skippednote) [inline components/button/Button.tsx:6]: let's use children instead

Comment (skippednote) [inline components/button/Button.tsx:6]: Text shouldn't have a default value

File: components/button/Button.tsx [hunk 4]
```diff
@@ -1,7 +1,50 @@
...
+  }
+  return (
+    <div
+      className={classes + ' ' + extraClasses}
+      onClick={() => (onClick != null ? onClick : null)}
+    >
+      {iconLeft && <img className="w-7" src={iconName} />}
+      <span className={textColor + ' mx-auto ' + textClasses}>{text}</span>
+      {iconRight && <img className="w-7" src={iconName} />}
+    </div>
```
Comment (skippednote) [inline components/button/Button.tsx:46]: Let's use the Icon component here

File: @types/button.types.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export type ButtonType = {
+  variant: string;
```
Comment (skippednote) [inline @types/button.types.ts:2]: Use the values instead.

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/button/Button.test.tsx (hunk 1 above)
Comment (tushar8184) [reply components/button/Button.test.tsx:1]: Made the required changes

File: components/button/Button.tsx (hunk 3 above)
Comment (tushar8184) [reply components/button/Button.tsx:6]: Done

Comment (tushar8184) [reply components/button/Button.tsx:6]: Done

File: components/button/Button.test.tsx (hunk 2 above)
Comment (tushar8184) [reply components/button/Button.test.tsx:13]: Have added one more test to check all the variants.

File: @types/button.types.ts (hunk 5 above)
Comment (tushar8184) [reply @types/button.types.ts:2]: Done

File: components/button/Button.test.tsx [hunk 6]
```diff
@@ -8,8 +8,38 @@ describe('Test Button component', () => {
...
+  it('renders the component based on variant', () => {
+    render(<Button variant="tertiary" />);
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('bg-light-off h-12');
+  });
+  it('renders the component based on variant', () => {
+    render(<Button variant="tertiary-small" />);
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('bg-light-off h-9');
+  });
```
Comment (skippednote) [inline components/button/Button.test.tsx:44]: Please add the variant name in the `it` string

File: @types/button.types.ts (hunk 5 above)
Comment (skippednote) [reply @types/button.types.ts:2]: You need to list all the variant names here instead of `string`

File: components/button/Button.test.tsx [hunk 7]
```diff
@@ -8,8 +8,38 @@ describe('Test Button component', () => {
     expect(button).toBeInTheDocument();
   });
   it('renders the correct child data', () => {
-    render(<Button>Test data</Button>);
+    render(<Button text="Test data"></Button>);
     const button = screen.getByRole('button');
     expect(button).toHaveTextContent(/test data/i);
   });
```
Comment (skippednote) [inline components/button/Button.test.tsx:14]: We need to add a test for `onClick`

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/button/Button.test.tsx [hunk 8]
```diff
@@ -1,15 +1,63 @@
...
+        onClick={(event) => {
+          event.currentTarget.classList.add('clicked');
+        }}
+      />,
+    );
+    const button = screen.getByRole('button');
+    fireEvent.click(button);
+    expect(button).toHaveClass('clicked');
+  });
+
```
Comment (skippednote) [inline components/button/Button.test.tsx:31]: Can we move this test to the bottom

File: components/button/Button.test.tsx [hunk 9]
```diff
@@ -1,15 +1,63 @@
...
   });
+
   it('renders the correct child data', () => {
-    render(<Button>Test data</Button>);
+    render(<Button variant="primary" text="Test data"></Button>);
     const button = screen.getByRole('button');
     expect(button).toHaveTextContent(/test data/i);
   });
+
+  it('display an alert onClick', () => {
```
Comment (skippednote) [inline components/button/Button.test.tsx:17]:
```suggestion
  it('add class on onClick', () => {
```

File: components/button/Button.tsx [hunk 10]
```diff
@@ -1,7 +1,48 @@
...
+  variant,
+  text,
+  iconLeft,
+  iconRight,
+  iconName,
+  extraClasses,
+  textClasses,
+  onClick,
+}: ButtonType) => {
+  let classes = '';
```
Comment (skippednote) [inline components/button/Button.tsx:14]:
```suggestion
  let classes = defaultClasses';
```

File: components/button/Button.tsx [hunk 11]
```diff
@@ -1,7 +1,48 @@
...
+  let textColor = '';
+
+  const defaultClasses =
+    'flex items-center w-full px-4 py-3 rounded-xs cursor-pointer';
+  if (variant === 'primary') {
+    classes = `${defaultClasses} bg-green-primary h-12`;
```
Comment (skippednote) [inline components/button/Button.tsx:20]:
```suggestion
    classes += ` bg-green-primary h-12`;
```

**Comment by skippednote** [comment]
Please remove the icons in this pr

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/button/Button.test.tsx (hunk 8 above)
Comment (tushar8184) [reply components/button/Button.test.tsx:31]: move to the bottom

File: components/button/Button.test.tsx (hunk 9 above)
Comment (tushar8184) [reply components/button/Button.test.tsx:17]: added 'add class on'

**Pull request merged**